Game/AI/Decks/LightswornShaddoldinosourExecutor.cs

```diff
@@ -1286,16 +1325,56 @@ private bool CoralDragoneff()
             }
             return false;
         }
+        public override void OnChaining(int player, ClientCard card)
+        {
+           // Logger.DebugWriteLine("current chain = " + Duel.CurrentChain.Count);
+           
+            if(ShieldTarget==null)
+            {
+                if (Enemy.HasInSpellZone(CardId.MoonMirrorShield))
+                {
+                    ShieldTarget = AI.Utils.GetChainTarget();
+                    
+                }
+                else
+                    ShieldTarget = null;
+            }          
+        
+        }
+        /* public override MainPhaseAction OnSelectIdleCmd(MainPhase main)
+         {
+
+             
+             return new MainPhaseAction(MainPhaseAction.MainAction.ToEndPhase);
+         }*/
 
         public override bool OnPreBattleBetween(ClientCard attacker, ClientCard defender)
         {
+            if (!Enemy.HasInSpellZone(CardId.MoonMirrorShield))
+            {
+                ShieldTarget = null;
+            }
             if (!defender.IsMonsterHasPreventActivationEffectInBattle())
             {
-                if (attacker.Id == CardId.ElShaddollConstruct && !attacker.IsDisabled()) // TODO: && defender.IsSpecialSummoned
-                    attacker.RealPower = 9999;
+                if (Enemy.HasInSpellZone(CardId.MoonMirrorShield))
+                {
+                    if (defender == ShieldTarget && defender.IsAttack())
+                        return false;
+                }
                 if (attacker.Id == CardId.UltimateConductorTytanno && !attacker.IsDisabled() && defender.IsDefense())
+                {
                     attacker.RealPower = 9999;
+                    return true;
```

Comment:
never return true in `OnPreBattleBetween` because more check should be done in `base.OnPreBattleBetween`